### DISCUSSION TITLE
buff throwing things

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#2571] [General]
https://github.com/cataclysmbnteam/Cataclysm-BN/blob/15ddc37a6cc7e55a94886ffdffddcdeeb27e1915/src/ranged.cpp#L1185-L1196

currently thrown damage is outrageously low. bring back the 1/2mv^2!
we should make an agreement on how the damage formula would look like first tho.

**Comment by yay855** [comment]
Strength should increase damage and range by a scaling amount, item weight should increase damage by a flat amount and *reduce* range by a scaling amount. The railgun should increase effective strength, but tbh I feel like it should also provide a flat damage bonus for metal items, so your character can effectively just throw ball bearings at foes with it turned on and actually kill.

**Comment by ghost** [comment]
A thing that is overlooked on this, what makes throwing terrible for me is the keypresses required. We need a way to throw held item in the case of holding dozens of rocks. Any other ideas to make it faster and less presses to throw items would be appreciated.

**Reply by scarf005** [reply]
some idea:
- `(t)hrow` default throws items held unless it's ranged weapons
- add a new hotkey, `Toggle throwable` for items

**Reply by KheirFerrum** [reply]
Suggest adding `throw held` as a new command with a combination command like Ctrl + T.

**Reply by ghost** [reply]
That would work, having throw and throw held means I can remove the keybind for throw and replace it with throw held item as I always wield what I throw, so that would fix keybindings as it's the same as press f to start aiming.

**Reply by Coolthulhu** [reply]
Most of the time, you probably want to throw types rather than individual items. That is, any rock, not rock just from the marked stack.

**Reply by KheirFerrum** [reply]
A way to set default ammo for throwing the same way as shooting would work.

**Reply by ratmice** [reply]
> A thing that is overlooked on this, what makes throwing terrible for me is the keypresses required. We need a way to throw held item in the case of holding dozens of rocks. Any other ideas to make it faster and less presses to throw items would be appreciated.

I have been thinking about this a little, the number of key presses gets even worse when trying to use throwing with things like `javelin_bag` and `bandolier_knife`. One option which I kind of liked was based on some kind of item tagging system.  There would then be 3 aspects to the system:

* A way to assign a tag to an item/stack of items, multiple stacks/items can share a tag.
* A way to assign a tag to a container, the contents of the container then get automatically filled/sheathed from tagged items (compatible with the container) on item pickup, and when wearing the container.
* A `throw_item_with_tag` like command to find an item which may involve removing it from the container.

This would essentially work similarly to the `= Reassign` of the inventory mechanism, except that multiple stacks would be able to share the assigned letter.  Rather than something automatically derived from item categorization.

There are a few more things that probably need some work, in particular:
`throwing` a weapon currently is possible even when using a 2h or a 1h weapon + shield, but `activating` a `bandolier` to `draw` a wields the weapon to hand.  This leads to some odd inconsistency: it is possible to throw while both hands are in use.
But from what I can see you cannot throw from a bandolier or javelin bag even when one hand is free.

Mostly I would lean towards keeping each thing that currently requires a keypress to be sort of a discrete step in the computing the AP cost but leveraging the tag system to reduce the number of keypresses involved.  Before I try and work on a PR I was curious to gauge interest in the idea.

**Reply by ZChris13** [reply]
take a que from DC:SS with the "quiver", where it's just marking a stack of items in your inventory as your preferred throwing weapons, with the controls to change it

**Reply by ratmice** [reply]
Yeah that is basically how nethack works, and it works well with those games.  My feeling is that it is a bit too simple for cataclysm, where you have things like bandoliers, as well as various stacks of different kinds of knives in inventory and in bandolier. A beginning throwing character is often throwing whatever items they can get their hands on, which rarely forms any nice stack until they can forge their own throwing weapons.

The tagging system is basically a more general form of this quiver mechanism, that allows multiple stacks to be assigned the quiver tag.  In theory the tag system is ordered so you could assign a bandolier to the first or last item.  Or throw from a stack of Throwing axes first and throwing knives last.

Essentially it breaks down this notion of quiver into like 3 stages:

* one mechanism to add arbitrary tags to items/stacks of items and holsters, for example one tag could be called `quiver`.
* A screen to adjust the ordering of stacks/items in a tag, from first to last.
* The assignment of an action like `throwing` to consume items from a `tag` before prompting the character for an item, such as assigning the `quiver` tag to the `throw` action.

I tried to sort of make it general, so that it can potentially be used for other things instead of being tied down specifically to throwing.  Really only the last item is specific to throwing.  I think another way the system could find use is in crafting.
If you allow specifying a tag for a place where items are stored, you could keep there a set of crafting ingredients that are designated for some particular use, but I haven't given much thought into how or whether it could/should be used in other ways.

**Comment by KheirFerrum** [comment]
As to strength and weight of object affecting damage. I'd have strength dictate the max weight of an object you can throw. The damage of the thrown object will then have a "base" damage based on it's weight (or thrown damage if that is defined), multiplied by a strength modifier.

We're going to need to tweak this a lot.

**Comment by scarf005** [comment]
resolved by #5088
